packages/storage/amplify_storage_s3_dart/lib/src/storage_s3_service/service/task/s3_upload_task.dart

```diff
@@ -721,22 +745,23 @@ class _CompletedSubtask {
 
 class _OngoingSubtask {
   _OngoingSubtask({
-    required this.partBodyChunk,
+    required Stream<List<int>> Function() partBodyGetter,
     required this.partNumber,
     required this.request,
-  });
+  }) : _partBodyGetter = partBodyGetter;
 
-  final List<int> partBodyChunk;
+  final Stream<List<int>> Function() _partBodyGetter;
   final int partNumber;
   final Future<void> request;
 
+  Stream<List<int>> get partBody => _partBodyGetter();
```

Comment:
Am I correct in understanding this would only get called once for this subtask? I just wasn't sure if there was some way a caller might generate 2 streams for the expected same result if called twice on same instance, in which case might make sense to memoize or change this somehow. Doesn't appear to be an issue currently, just making sure I'm not missing something.

Reply:
Thanks for the question. This is for making the part upload resumable. 

When pause upload operation, ongoing sub tasks will be canceled. The references of the sub tasks will be held in memory. When resume the upload tasks, it looks up in memory subtasks first and resume them if any, here we would need a new stream of the part content.